### PR TITLE
Create person

### PR DESCRIPTION
**Pull request by zemola**
- Create class `Person` with the following:

> Instance vars: `@id, @name, and @age.`
> Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
> Getters for @id, @name, and @age.
> Setters for @name and @age.
> Add private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
> Add public method can_use_services? that returns true if person is `of age` or if they have permission from parents.
> 

- Create class `Student` with the following:

> Inherits from` Person`.
> Constructor extends parent's constructor by adding `@classroom` and a parameter for it.
> Add method `play_hooky` that returns "¯\(ツ)/¯".

- Create class `Teacher` with the following:
> Inherits from Person.
> Constructor extends parent's constructor by adding @specialization and a parameter for it.
> Override can_use_services? so it always returns true.

- Each class is saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  public
+
+  def can_use_services?
+    of_age || parent_permission
```
Comment (mikethreels) [inline person.rb]: - [ ] there is a small typo in this method, this method is calling another method named `parent_permission` however it should check the value of the `@parent_permission` instance variable.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require './person'
+
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(classroom, age, name, parent_permission)
```
Comment (mikethreels) [inline student.rb:6]: - [ ] the `name` and `parent_permission` arguments should have a default value similar to in the `Person` class

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require './person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization, age, name, parent_permission)
```
Comment (mikethreels) [inline teacher.rb:6]: - [ ] the `name` and `parent_permission` arguments should have a default value similar to in the `Person` class

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Azeem, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BrianSammit — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @zemola, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 
- Great job working on the changes requested


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**